pkg/internalapis/cnsvolumeinfo/cnsvolumeinfoservice.go

```diff
@@ -283,14 +284,27 @@ func (volumeInfo *volumeInfo) GetVolumeInfoForVolumeID(ctx context.Context, volu
 }
 
 // PatchVolumeInfo patches the CNSVolumeInfo instance associated with volumeID in given parameters.
-func (volumeInfo *volumeInfo) PatchVolumeInfo(ctx context.Context, volumeID string, patchBytes []byte) error {
+func (volumeInfo *volumeInfo) PatchVolumeInfo(ctx context.Context, volumeID string, patchBytes []byte, allowedRetries int) error {
 	log := logger.GetLogger(ctx)
 
 	volumeInfoInstance, err := volumeInfo.GetVolumeInfoForVolumeID(ctx, volumeID)
 	if err != nil {
 		return logger.LogNewErrorf(log, "failed to fetch CnsVolumeInfo instance for volumeID: %q", volumeID)
 	}
-	return volumeInfo.k8sClient.Patch(ctx, volumeInfoInstance, client.RawPatch(types.MergePatchType, patchBytes))
+	attempt := 0
+	for {
+		attempt++
+		log.Infof("attempt %d, to patch cnsvolumeinfo", attempt)
+		err := volumeInfo.k8sClient.Patch(ctx, volumeInfoInstance, client.RawPatch(types.MergePatchType, patchBytes))
+		if err != nil && attempt >= allowedRetries {
+			log.Infof("attempt: %d, failed to patch the cnsvolumeinfo", attempt)
+			return err
+		} else if err == nil {
+			log.Infof("attempt: %d, Successfully patched the cnsvolumeinfo", attempt)
+		}
+		time.Sleep(100 * time.Millisecond)
```

Comment:
This is an infinite loop, if there are no error conditions, you should be able to break out.